packages/babel-core/src/transformation/pipeline.js

```diff
@@ -31,3 +38,50 @@ export function transformFromAst(ast: Object, code: string, opts: Object): Babel
     return file.transform();
   });
 }
+
+export function transformFile(filename: string, opts?: Object, callback: Function) {
+  if (typeof opts === "function") {
+    callback = opts;
+    opts = {};
+  }
+
+  opts.filename = filename;
+  opts = new OptionManager().init(opts);
```

Comment:
Pulled out `OptionManager#init` calls into these so that we won't have to bother reading the file content from disk if the filename fails the ignore/only check.